pkg/systemd/quadlet/quadlet.go

```diff
@@ -940,6 +942,12 @@ func ConvertNetwork(network *parser.UnitFile, name string, unitsInfoMap map[stri
 	// Need the containers filesystem mounted to start podman
 	service.Add(UnitGroup, "RequiresMountsFor", "%t/containers")
 
+	if network.LookupBooleanWithDefault(NetworkGroup, KeyNetworkDeleteOnStop, false) {
+		serviceStopPostCmd := createBasePodmanCommand(network, NetworkGroup)
+		serviceStopPostCmd.add("network", "rm", "--force", networkName)
```

Comment:
Sorry should have checked this earlier, do we really want force in this case? Per systemd ordering it will stopped after container are stopped but not when one runs `systemctl stop` manually on the network right?
It could be very unexpected if the removing the unit causes podman to force remove all container that are on the network.
Having the command error in case there still are other containers running might be a good thing.

Reply:
I tested this scenario: https://github.com/containers/podman/issues/23678#issuecomment-2310730806
From what I could tell, stopping the network unit will first stop the dependent container units and only then stop the network. So, the `force` flag is expected to be a noop as all containers are supposed to stop before the network is deleted. 

Having said that, it will remove containers that were added to this network without a proper connection in a Quadlet unit.

Reply:
Ack, thanks. So then I think it is even better to not add --force, anything else would mean we have a outside container on this network which is not managed by quadlet so I would prefer to not force remove that.
Returning an error in this case seems like the best option.

Reply:
Concur, force-removal seems too aggressive - likely to cause some collateral damage.

Reply:
OK, I removed the `--force` flag and also enhanced the test to make sure dependency works